src/python/WMCore/ReqMgr/CherryPyThreads/StepChainParentageFixTask.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import (division, print_function)
+
+from WMCore.REST.CherryPyPeriodicTask import CherryPyPeriodicTask
+from WMCore.Services.RequestDB.RequestDBWriter import RequestDBWriter
+from WMCore.Services.DBS.DBS3Reader import DBS3Reader
+
+
+def getChildDatasetsForStepChainMissingParent(reqmgrDB, status):
+    """
+    :param status: workflow status
+    :return: set of child dataset
+    """
+    results = reqmgrDB.getStepChainDatasetParentageByStatus(status)
+
+    childDatasets = set()
+    requestNames = set()
+    for reqName, info in results.items():
+        requestNames.add(reqName)
+        for dsInfo in info.values():
+            if dsInfo["ParentDset"]:
+                for childDS in dsInfo["ChildDsets"]:
+                    childDatasets.add(childDS)
+    return {"ChildDatasets": childDatasets, "Requests": requestNames}
+
+
+class StepChainParentageFixTask(CherryPyPeriodicTask):
+    """
+    Upldates StepChain parentage periodically
+    """
+
+    def __init__(self, rest, config):
+
+        super(StepChainParentageFixTask, self).__init__(config)
+        self.reqmgrDB = RequestDBWriter(config.reqmgrdb_url, logger=self.logger)
+        self.dbsSvc = DBS3Reader(config.dbs_url, logger=self.logger)
+        self.statusToCheck = ["announced", "normal-archived"]
+
+    def setConcurrentTasks(self, config):
+        """
+        sets the list of functions which
+        """
+        self.concurrentTasks = [{'func': self.fixStepChainParentage, 'duration': config.parentageFixDuration}]
+
+    def fixStepChainParentage(self, config):
+        """
+        Look through the stepchain workflows with ParentageResolved flag is False.
+        Fix the StepChain parentage and update the ParentageResolved flag to True
+        """
+        childDatasets = set()
+        requests = set()
+        for status in self.statusToCheck:
+            info = getChildDatasetsForStepChainMissingParent(self.reqmgrDB, status)
+            childDatasets = childDatasets.union(info["ChildDatasets"])
+            requests = requests.union(info["Requests"])
+
+        for childDS in childDatasets:
+            self.dbsSvc.fixMissingParentageDatasets(childDS, insertFlag=True)
```

Comment:
Then add another log like (check if self.logger is the correct one) 
```
msg = "Fixing dataset parentage for: %s" % childDS
msg += " Counter %d out of %d" % (dsetCount, len(childDatasets))
self.logger.info(msg)
```

such that we can monitor:
*) which datasets it fixes
*) roughly how long it takes
*) and how much work it has got in this cycle